File: repos_estudiantes_ALVAREZ_segunda_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/COMPOSICION-AGREGACION/EJERCICIO1/JAVA-EJERCICIO1/src/Ejercicio1/Ej1.java
package Ejercicio1;

import java.util.ArrayList;

public class Ej1 {
    public static void main(String[] args) {
        Casa casa = new Casa("Av. Siempre Viva 742");
        casa.agregar_habitacion(new Habitacion("Sala", 20));
        casa.agregar_habitacion(new Habitacion("Cocina", 12));
        casa.agregar_habitacion(new Habitacion("Dormitorio", 15));
        casa.agregar_habitacion(new Habitacion("Baño", 6));

        casa.mostrar_casa();
    }
}

class Habitacion {
    private String nombre;
    private int tamano;

    public Habitacion(String nombre, int tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public String get_nombre() {
        return nombre;
    }

    public void set_nombre(String nombre) {
        this.nombre = nombre;
    }

    public int get_tamano() {
        return tamano;
    }

    public void set_tamano(int tamano) {
        this.tamano = tamano;
    }

    public void mostrar_info() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}

class Casa {
    private String direccion;
    private ArrayList<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public String get_direccion() {
        return direccion;
    }

    public void set_direccion(String direccion) {
        this.direccion = direccion;
    }

    public ArrayList<Habitacion> get_habitaciones() {
        return habitaciones;
    }

    public void agregar_habitacion(Habitacion h) {
        habitaciones.add(h);
    }

    public void mostrar_casa() {
        System.out.println("Dirección de la casa: " + direccion);
        System.out.println("Habitaciones:");
        for (Habitacion h : habitaciones) {
            h.mostrar_info();
        }
    }
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/COMPOSICION-AGREGACION/EJERCICIO1/Ej1.py
class Habitacion:
    def __init__(self, nombre, tamano):
        self.__nombre = nombre
        self.__tamano = tamano  

    def get_nombre(self):
        return self.__nombre

    def get_tamano(self):
        return self.__tamano

    def set_nombre(self, nombre):
        self.__nombre = nombre

    def set_tamano(self, tamano):
        self.__tamano = tamano

    def mostrar_info(self):
        print(f"Habitación: {self.__nombre}, Tamaño: {self.__tamano} m²")


class Casa:
    def __init__(self, direccion):
        self.__direccion = direccion
        self.__habitaciones = []

    def get_direccion(self):
        return self.__direccion

    def get_habitaciones(self):
        return self.__habitaciones

    def set_direccion(self, direccion):
        self.__direccion = direccion

    def agregar_habitacion(self, habitacion):
        self.__habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"Dirección de la casa: {self.__direccion}")
        print("Habitaciones:")
        for habitacion in self.__habitaciones:
            habitacion.mostrar_info()

casa = Casa("Av. Siempre Viva 742")
casa.agregar_habitacion(Habitacion("Sala", 20))
casa.agregar_habitacion(Habitacion("Cocina", 12))
casa.agregar_habitacion(Habitacion("Dormitorio", 15))
casa.agregar_habitacion(Habitacion("Baño", 6))

casa.mostrar_casa()